pkg/operation/controlplanesecrets/secrets_manager.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controlplanesecrets
+
+import (
+	gardencorev1alpha1helper "github.com/gardener/gardener/pkg/apis/core/v1alpha1/helper"
+	"github.com/gardener/gardener/pkg/utils/secrets"
+)
+
+// SecretConfigGeneratorFunc is a function which can generate a list of secret configurations
+// which could require a reference to a BasicAuth secret, StaticToken secret and certificate authorities.
+type SecretConfigGeneratorFunc func(basicAuthAPIServer *secrets.BasicAuth, staticToken *secrets.StaticToken, certificateAuthorities map[string]*secrets.Certificate) ([]secrets.ConfigInterface, error)
+
+// SecretsManager contains secret configurations which are used to generate and deploy secrets in the Shoot's Control Plane
+// It will save all generated secrets InfoData into the passed GardenerResourceDataList
+type SecretsManager struct {
+	GardenerResourceDataList gardencorev1alpha1helper.GardenerResourceDataList
+
+	APIServerBasicAuthConfig    secrets.ConfigInterface
+	StaticTokenConfig           secrets.ConfigInterface
+	CertificateAuthorityConfigs map[string]*secrets.CertificateSecretConfig
+	SecretConfigGenerator       SecretConfigGeneratorFunc
+
+	CertificateAuthorities map[string]*secrets.Certificate
+}
+
+// LoadCertificateAuthorities will load the CertificateAuthorities from the available GardenerResourceDataList
+func (s *SecretsManager) LoadCertificateAuthorities() error {
```

Comment:
Now that you have, very exemplarily, added unit tests for most new functions, can you also add one for this function?